website_product_subscription/controllers/subscribe_form.py

```diff
@@ -71,73 +70,60 @@ def normalize_form_data(self):
         else:
             self.qcontext["is_gift"] = False
 
+    def _validate_email_format(self, email):
+        if not tools.single_email_re.match(email):
```

Comment:
as far as I know this could be done a the form level setting the right type on the input field.